src/app/api/v1/endpoints/extract_data.py

```diff
@@ -0,0 +1,290 @@
+    NotFoundResponse,
+)
+from stratigraphy.coordinates.coordinate_extraction import CoordinateExtractor, LV03Coordinate, LV95Coordinate
+from stratigraphy.elevation.elevation_extraction import ElevationExtractor
+from stratigraphy.util.extract_text import extract_text_lines_from_bbox
+
+
+def extract_data(extract_data_request: ExtractDataRequest) -> ExtractDataResponse:
+    """Extract information from PNG images.
+
+    Args:
+        extract_data_request (ExtractDataRequest): The request data.
+
+    Returns:
+        ExtractDataResponse: The extracted information.
+    """
+    # Load the PNG image
+    pdf_document = load_pdf_from_aws(extract_data_request.filename)
+
+    # Load the page from the PDF document
+    pdf_page = pdf_document.load_page(extract_data_request.page_number - 1)
+    pdf_page_width = pdf_page.rect.width
+    pdf_page_height = pdf_page.rect.height
+
+    # Load the PNG image the boreholes app is showing to the user
+    # Convert the PDF filename to a PNG filename: "pdfs/geoquat/train/10012.pdf" -> 'pngs/geoquat/train/10012_0.png'
+    # Remove the file extension and replace it with '.png'
+    base_filename = extract_data_request.filename.rsplit(".", 1)[0]
+    png_filename = f"{base_filename}-{extract_data_request.page_number}.png"
+
+    # Load the PNG file from AWS
+    png_page = load_png_from_aws(png_filename)
+    png_page_width = png_page.shape[1]
+    png_page_height = png_page.shape[0]
+
+    # Convert the bounding box to the PDF coordinates
+    user_defined_bbox = extract_data_request.bbox.rescale(
+        original_height=png_page_height,
+        original_width=png_page_width,
+        target_height=pdf_page_height,
+        target_width=pdf_page_width,
+    )  # bbox in PDF coordinates
+
+    # Extract the information based on the format type
+    if extract_data_request.format == FormatTypes.COORDINATES:
+        # Extract the coordinates and bounding box
+        extracted_coords: ExtractCoordinatesResponse | None = extract_coordinates(
+            extract_data_request, pdf_page, user_defined_bbox.to_fitz_rect()
+        )
+
+        if extracted_coords is None:
+            return NotFoundResponse(
+                detail="Coordinates not found.",
+                bbox=extract_data_request.bbox,
+            )
+
+        # Convert the bounding box to PNG coordinates and return the response
+        return ExtractCoordinatesResponse(
+            bbox=extracted_coords.bbox.rescale(
+                original_height=pdf_page_height,
+                original_width=pdf_page_width,
+                target_height=png_page_height,
+                target_width=png_page_width,
+            ),
+            coordinates=Coordinates(
+                east=extracted_coords.coordinates.east,
+                north=extracted_coords.coordinates.north,
+                projection=extracted_coords.coordinates.projection,
+            ),
+        )
+    elif extract_data_request.format == FormatTypes.ELEVATION:
```

Comment:
I'm starting to question whether we need "elevation" as a dedicated format in this API at all. Wouldn't "number" do the trick as well as far as the "intelligent annotation" use case is concerned, as long as the user draws that bounding box precisely enough (i.e. so that no other numbers are inside the box)?

Reply:
I see your point. The extraction of elevation using the elevation option might be more robust which could be a nice value add. However, I do not see any reason why the number interface would do the job (if the bbox is drawn correctly). Let me know your preference. If we consider the "intelligent annotation" use case, I guess it would make more sense to remove it.

Reply:
Let's remove it for now.